spyder_kernels/customize/spydercustomize.py

```diff
@@ -569,15 +606,17 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
             else:
                 _print("Working directory {} doesn't exist.\n".format(wdir))
 
-        if __umr__.has_cython:
-            # Cython files
-            with io.open(filename, encoding='utf-8') as f:
-                ipython_shell.run_cell_magic('cython', '', f.read())
-        else:
-            exec_code(file_code, filename, ns_globals, ns_locals,
-                      post_mortem=post_mortem)
-
-        sys.argv = ['']
+        try:
```

Comment:
Why is this closed in a `try` now?

Reply:
Because in case of failure `sys.argv` is not reset. This was an undiscovered but that I am fixing here.